data-structures/Heap.cpp

```diff
@@ -0,0 +1,84 @@
+#include<iostream>
+#include<vector>
+using namespace std;
+
+class Heap{
+	private: 
+	int TYPE; // MIN => 0, MAX => 1
+	vector<int> vec;
+
+	public:
+	Heap(int TYPE){
+		this->TYPE = TYPE;
+	}
+
+	void insert(int num){
+		this->vec.push_back(num);
+		int i = this->vec.size() - 1;
+		if(this->TYPE == 0){
+			while(i>0 && num < this->vec[(i-1)/2]){
+				this->vec[i] = this->vec[(i-1)/2];
+				i = (i-1)/2;
+			}
+		}
+		else{
+			while(i>0 && num > this->vec[(i-1)/2]){
```

Comment:
Please use this code style conventions for loops https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements